// @handsontable/angular/esm2022/lib/hot-column.component.mjs
import { Component, Input, } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./hot-table.component";
export class HotColumnComponent {
    parentComponent;
    firstRun = true;
    // handsontable column options
    allowEmpty;
    allowHtml;
    allowInvalid;
    checkedTemplate;
    className;
    columnSorting;
    colWidths;
    commentedCellClassName;
    copyable;
    correctFormat;
    data;
    dateFormat;
    defaultDate;
    editor;
    filteringCaseSensitive;
    headerClassName;
    invalidCellClassName;
    label;
    language;
    noWordWrapClassName;
    numericFormat;
    placeholder;
    placeholderCellClassName;
    readOnly;
    readOnlyCellClassName;
    renderer;
    selectOptions;
    skipColumnOnPaste;
    sortByRelevance;
    source;
    strict;
    title;
    trimDropdown;
    type;
    uncheckedTemplate;
    validator;
    visibleRows;
    width;
    wordWrap;
    constructor(parentComponent) {
        this.parentComponent = parentComponent;
    }
    ngOnInit() {
        this.firstRun = false;
        this.parentComponent.addColumn(this);
    }
    ngOnChanges() {
        if (this.firstRun) {
            return;
        }
        this.parentComponent.onAfterColumnsChange();
    }
    ngOnDestroy() {
        this.parentComponent.removeColumn(this);
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.11", ngImport: i0, type: HotColumnComponent, deps: [{ token: i1.HotTableComponent }], target: i0.ɵɵFactoryTarget.Component });
    static ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.11", type: HotColumnComponent, selector: "hot-column", inputs: { allowEmpty: "allowEmpty", allowHtml: "allowHtml", allowInvalid: "allowInvalid", checkedTemplate: "checkedTemplate", className: "className", columnSorting: "columnSorting", colWidths: "colWidths", commentedCellClassName: "commentedCellClassName", copyable: "copyable", correctFormat: "correctFormat", data: "data", dateFormat: "dateFormat", defaultDate: "defaultDate", editor: "editor", filteringCaseSensitive: "filteringCaseSensitive", headerClassName: "headerClassName", invalidCellClassName: "invalidCellClassName", label: "label", language: "language", noWordWrapClassName: "noWordWrapClassName", numericFormat: "numericFormat", placeholder: "placeholder", placeholderCellClassName: "placeholderCellClassName", readOnly: "readOnly", readOnlyCellClassName: "readOnlyCellClassName", renderer: "renderer", selectOptions: "selectOptions", skipColumnOnPaste: "skipColumnOnPaste", sortByRelevance: "sortByRelevance", source: "source", strict: "strict", title: "title", trimDropdown: "trimDropdown", type: "type", uncheckedTemplate: "uncheckedTemplate", validator: "validator", visibleRows: "visibleRows", width: "width", wordWrap: "wordWrap" }, usesOnChanges: true, ngImport: i0, template: '', isInline: true });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.11", ngImport: i0, type: HotColumnComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'hot-column',
                    template: '',
                }]
        }], ctorParameters: () => [{ type: i1.HotTableComponent }], propDecorators: { allowEmpty: [{
                type: Input
            }], allowHtml: [{
                type: Input
            }], allowInvalid: [{
                type: Input
            }], checkedTemplate: [{
                type: Input
            }], className: [{
                type: Input
            }], columnSorting: [{
                type: Input
            }], colWidths: [{
                type: Input
            }], commentedCellClassName: [{
                type: Input
            }], copyable: [{
                type: Input
            }], correctFormat: [{
                type: Input
            }], data: [{
                type: Input
            }], dateFormat: [{
                type: Input
            }], defaultDate: [{
                type: Input
            }], editor: [{
                type: Input
            }], filteringCaseSensitive: [{
                type: Input
            }], headerClassName: [{
                type: Input
            }], invalidCellClassName: [{
                type: Input
            }], label: [{
                type: Input
            }], language: [{
                type: Input
            }], noWordWrapClassName: [{
                type: Input
            }], numericFormat: [{
                type: Input
            }], placeholder: [{
                type: Input
            }], placeholderCellClassName: [{
                type: Input
            }], readOnly: [{
                type: Input
            }], readOnlyCellClassName: [{
                type: Input
            }], renderer: [{
                type: Input
            }], selectOptions: [{
                type: Input
            }], skipColumnOnPaste: [{
                type: Input
            }], sortByRelevance: [{
                type: Input
            }], source: [{
                type: Input
            }], strict: [{
                type: Input
            }], title: [{
                type: Input
            }], trimDropdown: [{
                type: Input
            }], type: [{
                type: Input
            }], uncheckedTemplate: [{
                type: Input
            }], validator: [{
                type: Input
            }], visibleRows: [{
                type: Input
            }], width: [{
                type: Input
            }], wordWrap: [{
                type: Input
            }] } });
