// @handsontable/angular/esm2022/lib/hot-table-registerer.service.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
const instances = new Map();
export const HOT_DESTROYED_WARNING = 'The Handsontable instance bound to this component was destroyed and cannot be' +
    ' used properly.';
export class HotTableRegisterer {
    getInstance(id) {
        const hotInstance = instances.get(id);
        if (hotInstance.isDestroyed) {
            console.warn(HOT_DESTROYED_WARNING);
            return null;
        }
        return hotInstance;
    }
    registerInstance(id, instance) {
        return instances.set(id, instance);
    }
    removeInstance(id) {
        return instances.delete(id);
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.11", ngImport: i0, type: HotTableRegisterer, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
    static ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.11", ngImport: i0, type: HotTableRegisterer });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.11", ngImport: i0, type: HotTableRegisterer, decorators: [{
            type: Injectable
        }] });
